FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_ARP.c

```diff
@@ -583,7 +601,11 @@ NetworkBufferDescriptor_t *pxNetworkBuffer;
 		if( xIsCallingFromIPTask() != 0 )
 		{
 			/* Only the IP-task is allowed to call this function directly. */
-			xNetworkInterfaceOutput( pxNetworkBuffer, pdTRUE );
+			if( xNetworkInterfaceOutput( pxNetworkBuffer, pdTRUE ) != pdTRUE )
+			{
+				/* Not sent? Bad news. Maybe link is down? */
+				FreeRTOS_printf( ( "xNetworkInterfaceOutput failed. Link down?\n" ) );
```

Comment:
I don't know if we want to use a general `FreeRTOS_printf()` here, since this could produce a lot of console output in normal operation.  Since we don't have a network interface metrics object (that I am aware of), for now, let's just make sure that debugging output has been configured in the build.  See [example here.](https://github.com/FreeRTOS/FreeRTOS/blob/master/FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_ARP.c#L646)

```c
if( xNetworkInterfaceOutput( pxNetworkBuffer, pdTRUE ) != pdTRUE )
{
#if( ipconfigHAS_PRINTF != 0 ) || ( ipconfigHAS_DEBUG_PRINTF != 0 )
        FreeRTOS_printf( ( "xNetworkInterfaceOutput failed. Link down?\n" ) 
#else 
        ;   /* No action */
#endif
}
```

Reply:
Done. I did not add the #else though. I hope its fine?

Reply:
I did not dig into how ```FreeRTOS_printf()``` is implemented, apologies for the ignorance there. 

Ideally, there should be some logger component, in which we could configure:
- which level of logging a module wants. (If +TCP wants verbose logging, then we could just configure it.) 
- where things log to. (could have multiple stream. UART, file, ...)
So the caller could avoid all the #if-#else blocks.

This implementation is much better than where we started, that I'm willing to get things going. Comment only, no change request.